source/common/utility.h

```diff
@@ -54,6 +55,15 @@ class Utility {
    * @param argv forwarded argv argument of the main entry point.
    */
   static void parseCommand(TCLAP::CmdLine& cmd, const int argc, const char* const* argv);
+
+  /**
+   * @param host_port host:port as a string, where host can be IPv4, [IPv6], or a DNS
+   * name.
+   * @param host string* to receive the host if the parse succeeds
+   * @param port int* to receive the port if the parse succeeds
+   * @return bool true if the input could be parsed as host:port
+   */
+  static bool parseHostPort(const std::string& host_port, std::string* host, int* port);
```

Comment:
Can we add a unit test for this function exercising the various behaviors? It looks like a good candidate (well isolated) and the unit test would help in documenting what it is and isn't supposed to do.